### PR TITLE
Fix global layout issues

### LINKED ISSUE #825: About page is broken
## Description
	
CleanShot 2022-02-07 at 21 48 41@2x

Due to what looks like a mysterious empty `div` floating in the page, the css grid styles for the primary Openverse page layout are broken on the about page.

- https://search-staging.openverse.engineering/about

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes #825 

## Description

This PR makes some cleanup to the main layout logic of every page:

- Ensures the header and main content area are the first two child elements of our primary grid container, by grouping the locale banner, cc banner, and header into a div. 
- Only applies our grid layout above the medium breakpoint
- Removes a hardcoded `min-width: 300px` from the `html` element
- Uses viewport width units to fix some flexbox behavior in the header, where using `100%` width allowed the header to grow beyond the device width.
- Hides header logo text and content switcher label on _extremely_ small screens via a new `xs` breakpoint. Time to make Openverse work on smart watches 😄 
- **Unrelated bugfix**: Localizes the header logo link properly 

In addition to fixing the layout bugs, this should put us in a better position while @panchovm works on additional header revisions.

## Testing Instructions

View on an actual mobile device and by resizing your desktop browser window.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by sarayourfriend — COMMENTED** [review]
Code looks good so far and seems to work fine on desktop. Getting weird message-less and stack-trace-less errors on mobile browser though, so will continue debugging to see if I can find the cause of that.

**Comment by sarayourfriend** [comment]
Still getting the same mysterious error 😞 It prevents the JS on the page from running at all.

**Comment by sarayourfriend** [comment]
It's the same on Firefox for Android and mobile Safari 🤔 I'll keep trying to debug.

**Comment by sarayourfriend** [comment]
It appears to be an issue on `main` actually.

File: src/components/VHeader/VLogoButton.vue [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <template>
   <VButton
     as="NuxtLink"
-    to="/"
+    :to="localePath('/')"
```
Comment (obulat) [inline src/components/VHeader/VLogoButton.vue:4]: Great catch!

File: src/layouts/default.vue [hunk 2]
```diff
@@ -108,23 +110,30 @@ export default embeddedPage
   grid-template-rows: auto 1fr;
 }
 
+@screen md {
+  // Logic for displaying the filter sidebar and search results
+  // as independently-scrolling sections.
+  .main {
```
Comment (krysal) [inline src/layouts/default.vue:116]: Thanks for documenting the grid logic!

**Review by krysal — APPROVED** [review]
Since the issue that Sara mentioned is unrelated to these changes this LGTM 👍

File: src/layouts/default.vue [hunk 3]
```diff
@@ -108,23 +110,30 @@ export default embeddedPage
...
+
 .main {
-  display: grid;
-  grid-template-columns: 1fr 316px;
-  height: 100%;
   overflow: hidden;
 }
-
-.main > * {
+.main > *:not(:empty) {
```
Comment (obulat) [inline src/layouts/default.vue:135]: Adding `:not(:empty)` is such a great solution to all the teleports taking up grid areas when empty!

**Review by obulat — APPROVED** [review]
The layout works perfectly!

**Pull request merged**